sda-admin/helpers/helpers.go

```diff
@@ -0,0 +1,166 @@
+package helpers
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"regexp"
+	"strings"
+	"strconv"
+	"time"
+
+	"github.com/golang-jwt/jwt"
+)
+
+// Helper functions used by more than one module
+
+// CheckTokenExpiration is used to determine whether the token is expiring in less than a day
+func CheckTokenExpiration(accessToken string) error {
+	// Parse jwt token with unverifies, since we don't need to check the signatures here
+	token, _, err := new(jwt.Parser).ParseUnverified(accessToken, jwt.MapClaims{})
+	if err != nil {
+		return fmt.Errorf("could not parse token, reason: %s", err)
+	}
+
+	claims, ok := token.Claims.(jwt.MapClaims)
+	if !ok {
+		return fmt.Errorf("broken token (claims are empty): %v\nerror: %s", claims, err)
+	}
+
+	// Check if the token has exp claim
+	if claims["exp"] == nil {
+		return fmt.Errorf("could not parse token, reason: no expiration date")
+	}
+
+	// Parse the expiration date from token, handle cases where
```

Comment:
Yes. In case of "bad token", the message is shown in the log but the user will not see it. It's better to check it on the client side as well.

Reply:
This is a CLI, it doesn't have logs. Either it is shown to the user or it is not relevant.

Reply:
I mean the logs of the container. The reason about the problem of bad token is only shown in the logs of the container and not returned from the API call. That's why `CheckTokenExpiration` needs to be called in the CLI.

Reply:
Aha, I see your point @nanjiangshu. It would be nice if the API call would return the status of the token, so that we don't have to duplicate that code. But I agree that the most important thing is that the users understand what's happening and gets an informative message if the token is bad.

Reply:
Yes, the API has its short comings, but we shouldn't implement band aids here for things that needs to be fixed in the API.

Reply:
in order for the user to know what went wrong you need to take action depending on the response you get back from the API.

No token, bad token and expired token will all generate a response status code of `401`.
The actual response body will differ depending on the case:
* No token: "no access token supplied"
* Bad token and expired token: message starts with "signed token not valid"

Reply:
make sense. I have remove token checking from the CLI and show the response when server returned error. 
Now the error message looks informative 
```
error: failed to ingest file, reason: server returned status 401: {"error":"signed token not valid: invalid JWT, (token was 11)"}
```
implemented in the commits 575ddea and 7bfac1e